modin/core/dataframe/algebra/groupby.py

```diff
@@ -373,16 +412,196 @@ def try_filter_dict(agg_func, df):
             Aggregation function.
         df : pandas.DataFrame
             Serialized partition which contains available columns.
+        preserve_aggregation_order : bool, default: True
+            Whether to manually restore the order of columns for the result specified
+            by the `agg_func` keys (only makes sense when `agg_func` is a dictionary).
 
         Returns
         -------
-        Callable
+        callable(pandas.core.groupby.DataFrameGroupBy) -> pandas.DataFrame
             Aggregation function that can be safely applied to this particular partition.
         """
         if not isinstance(agg_func, dict):
             return agg_func
-        partition_dict = {k: v for k, v in agg_func.items() if k in df.columns}
-        return lambda grp: grp.agg(partition_dict)
+
+        grp_has_id_level = df.columns.names[0] == cls.ID_LEVEL_NAME
+        # The 'id' level prevents us from a lookup for the original
+        # partition's columns. So dropping the level.
+        partition_columns = df.columns.droplevel(0) if grp_has_id_level else df.columns
+
+        partition_dict = {k: v for k, v in agg_func.items() if k in partition_columns}
+        return cls._build_callable_for_dict(
+            partition_dict, preserve_aggregation_order, grp_has_id_level
+        )
+
+    @classmethod
+    def _build_callable_for_dict(
+        cls, agg_dict, preserve_aggregation_order=True, grp_has_id_level=False
+    ):
+        """
+        Build callable for an aggregation dictionary.
+
+        Parameters
+        ----------
+        agg_dict : dict
+            Aggregation dictionary.
+        preserve_aggregation_order : bool, default: True
+            Whether to manually restore the order of columns for the result specified
+            by the `agg_func` keys (only makes sense when `agg_func` is a dictionary).
+        grp_has_id_level : bool, default: False
+            Whether the frame we're grouping on has intermediate columns
+            (see ``cls.ID_LEVEL_NAME``).
+
+        Returns
+        -------
+        callable(pandas.core.groupby.DataFrameGroupBy) -> pandas.DataFrame
+        """
+        external_aggs = {}
+        native_aggs = {}
+
+        result_columns = []
+        for col, func, func_name, col_renaming_required in walk_aggregation_dict(
+            agg_dict
+        ):
+            # Filter dictionary
+            dict_to_add = (
+                external_aggs if cls.is_external_aggregation(func) else native_aggs
+            )
+
+            new_value = func if func_name is None else (func_name, func)
+            old_value = dict_to_add.get(col)
+
+            if old_value is not None:
+                assert isinstance(
+                    old_value, list
+                ), f"`{col_renaming_required=}` must be `True` when multiple aggregations per columns is specified."
```

Comment:
assert check and message do not look related to each other, why is that?

Reply:
right, the message is a bit confusing, replaced it with the following:
```python
                ErrorMessage.catch_bugs_and_request_email(
                    failure_condition=not isinstance(old_value, list),
                    extra_log="Expected for all aggregation values to be a list when at least "
                    + f"one column has multiple aggregations. Got: {old_value} {type(old_value)}",
                )
```